sdks/python/setup.py

```diff
@@ -195,7 +195,7 @@ def get_version():
     'google-cloud-spanner>=1.13.0,<2',
     'grpcio-gcp>=0.2.2,<1',
     # GCP Packages required by ML functionality
-    'google-cloud-dlp>=0.12.0,<4',
+    'google-cloud-dlp>=3.3.1,<4',
```

Comment:
Consider re-running `./gradlew :sdks:python:generatePythonRequirementsAll` for this (potentially in a follow-up if you'd rather get tests green now). CC: @tvalentyn

Reply:
The requirements seem already have 3.3.1.

Reply:
We did regenerate requirements for dlp already. I think we didn't catch this issue in https://github.com/apache/beam/pull/15988 because Dataflow containers still included an older version of DLP.

Reply:
so integration tests passed on that PR

Reply:
it would be good to identify the lower bound, and CP this into RC3. perhaps 3.0 ?

Reply:
FYI, this is the changelog: https://github.com/googleapis/python-dlp/blob/main/CHANGELOG.md

Reply:
looks like it is removed from https://github.com/googleapis/python-dlp/commit/c6001e20facb0bba957794c674c7b1121dc1774a#diff-1731e2aca42029b3a9b5cca07e2a64822f03b2da6ad666568a6918a424f764fe, and common_project_path is added https://github.com/googleapis/python-dlp/commit/4f3148e93ec3dfc9395aa38a3afc62498500a055#. 3.0.0 is the lower bound.

Reply:
oh, I assumed you tracked it down and the breaking change was introduced in 3.3.1, agree we should make this google-cloud-dlp>=3,<4

Reply:
thanks for tracking it down. sending https://github.com/apache/beam/pull/16164 and https://github.com/apache/beam/pull/16165